pipeline_tools/tests/test_create_analysis_metadata.py

```diff
@@ -180,10 +180,12 @@ def test_get_workflow_tasks(self, data_file):
 
     def test_get_file_format(self):
         assert cam.get_file_format('asdf', {}) == 'unknown'
-        assert cam.get_file_format('asdf.bam', {'.bam': 'bam'}) == 'bam'
-        assert cam.get_file_format('asdf.txt', {'.bam': 'bam'}) == 'unknown'
-        assert cam.get_file_format('asdf.bam', {'.bam': 'bam', '_metrics': 'metrics'}) == 'bam'
-        assert cam.get_file_format('asdf.foo_metrics', {'.bam': 'bam', '_metrics': 'metrics'}) == 'metrics'
+        assert cam.get_file_format('asdf.bam', {'.bam$': 'bam'}) == 'bam'
+        assert cam.get_file_format('asdf.txt', {'.bam$': 'bam'}) == 'unknown'
```

Comment:
We may not have this situation with the SS2 outputs, but what if a file matches multiple patterns? E.g. `test_metrics.txt`

Reply:
With `$` in the regex, we won't run into that situation. For example, `test_metrics.txt` will only be matched by `.txt$`